pkg/provisionclients/swagger/nodebootstrapping.json

```diff
@@ -0,0 +1,653 @@
+          "type": "integer",
+          "format": "int32",
+          "x-nullable": true
+        },
+        "messageOfTheDay": {
+          "type": "string",
+          "x-nullable": true
+        },
+        "architecture": {
+          "type": "string",
+          "x-nullable": true
+        },
+        "agentPoolWindowsProfile": {
+          "$ref": "#/definitions/AgentPoolWindowsProfile",
+          "x-nullable": true
+        },
+        "securityProfile": {
+          "$ref": "#/definitions/AgentPoolSecurityProfile",
+          "x-nullable": true
+        },
+        "gpuProfile": {
+          "$ref": "#/definitions/GPUProfile",
+          "x-nullable": true
+        },
+        "artifactStreamingProfile": {
+          "$ref": "#/definitions/ArtifactStreamingProfile",
+          "x-nullable": true
+        },
+        "nodeInitializationTaints": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "x-nullable": true
+        },
+        "nodeTaints": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "x-nullable": true
+        },
+        "maxPods": {
+          "type": "integer",
+          "format": "int32",
+          "x-nullable": true
+        }
+      },
+      "required": [
+        "name",
+        "vmSize",
+        "osType",
+        "osSku",
+        "storageProfile",
+        "distro",
+        "orchestratorVersion",
+        "vnetCidrs",
+        "vnetSubnetID",
+        "mode",
+        "architecture",
+        "maxPods"
+      ]
+    },
+    "CustomKubeletConfig": {
+      "type": "object",
+      "properties": {
+        "cpuManagerPolicy": {
+          "type": "string",
+          "x-nullable": true
+        },
+        "cpuCfsQuota": {
+          "type": "boolean",
+          "x-nullable": true
+        },
+        "cpuCfsQuotaPeriod": {
+          "type": "string",
+          "x-nullable": true
+        },
+        "imageGcHighThreshold": {
+          "type": "integer",
+          "format": "int32",
+          "x-nullable": true
+        },
+        "imageGcLowThreshold": {
+          "type": "integer",
+          "format": "int32",
+          "x-nullable": true
+        },
+        "topologyManagerPolicy": {
+          "type": "string",
+          "x-nullable": true
+        },
+        "allowedUnsafeSysctls": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "x-nullable": true
+        },
+        "failSwapOn": {
+          "type": "boolean",
+          "x-nullable": true
+        },
+        "containerLogMaxSizeMB": {
+          "type": "integer",
+          "format": "int32",
+          "x-nullable": true
+        },
+        "containerLogMaxFiles": {
+          "type": "integer",
+          "format": "int32",
+          "x-nullable": true
+        },
+        "podMaxPids": {
+          "type": "integer",
+          "format": "int32",
```

Comment:
Note that actual kubelet `podPidsLimit` is int64. Something we likely need to fix on the AKS side, might as well type it correctly here.

Reply:
I think we should change ours after to not misalign with AKS requirements...